### ISSUE TITLE
Federated Dropout for Federated Learning

### ISSUE THREAD

**Issue by fanto88** [tensorflow/federated#2005]
Hi all, i'm not sure this is the correct place to ask this.

I'm doing a personal research for Tensorflow Federated and i was really interested in the idea of Federated Dropout, so basically i give the client a smaller model to train and then, on the server side, i put back all the update to the original model. This idea came from  Adaptive Federated Dropout: Improving Communication Efficiency and Generalization for Federated Learning
Nader Bouacida, Jiahui Hou, Hui Zang, Xin Liu.

I understood the idea behind, but i can't see how to implement it on the simple_fedavg example. Any suggestion on how to do so?
